hera_sim/simulate.py

```diff
@@ -0,0 +1,335 @@
+                time_per_integ=time_per_integ,
+                min_freq=min_freq,
+                channel_bw=channel_bw,
+                instrument=instrument,
+                telescope_location=telescope_location,
+                telescope_lat_lon_alt=telescope_lat_lon_alt,
+                object_name=object_name,
+                start_jd=start_jd,
+                vis_units=vis_units,
+            )
+
+        else:
+            self.data = self._read_data(self.data_filename, pols, **kwargs)
+
+    def _read_data(self, filename, polarizations, **kwargs):
+        uv = UVData()
+        uv.read(filename, polarizations=polarizations, read_data=True, **kwargs)
+        return uv
+
+    def write_data(self, filename, file_type=None, **kwargs):
+        """
+        Write current UVData object to file.
+
+        Args:
+            filename (str): filename to write to. Suffix of filename will determine the type of file to be written,
+                unless `file_type` is set.
+            file_type: (str): one of "miriad", "uvfits" or "uvh5" (i.e. any of the supported write methods of
+                :class:`pyuvdata.UVData`) which determines which write method to call.
+            **kwargs: keyword arguments sent directly to the write method chosen.
+        """
+        if file_type is None:
+            try:
+                file_type = filename.split(".")[-1]
+            except IndexError:
+                raise ValueError(
+                    "if no file_type is given, the filename must be suffixed with either .miriad, .uvfits or .uvh5"
+                )
+
+        if file_type not in ["miriad", "uvfits", "uvh5"]:
+            raise ValueError("file_type must be one of 'miriad', 'uvfits' or 'uvh5'")
+
+        getattr(self.data, "write_%s" % filename.split(".")[-1])(filename, **kwargs)
+
+    def _iterate_baselines(self, with_conj=True):
+        """
+        Iterate through baselines in the data object
+        """
+        for i, ant1 in enumerate(self.data.antenna_numbers):
+            for j, ant2 in enumerate(
+                self.data.antenna_numbers[(i + 1 if with_conj else 0) :]
+            ):
+                # Get the Nblts indices of this baseline (and its conjugate)
+                yield ant1, ant2, self.data.antpair2ind(
+                    ant1, ant2, ordered=not with_conj
+                )
+
+    @_model()
+    def add_eor(self, model, **kwargs):
+        """
+        Add an EoR-like model to the visibilities.
+
+        Args:
+            model (str or callable): either a string name of a model function existing in :mod:`~hera_sim.eor`, or
+                a callable which has the signature ``fnc(lsts, fqs, bl_len_ns, **kwargs)``.
+            **kwargs: keyword arguments sent to the EoR model function, other than `lsts`, `fqs` and `bl_len_ns`.
+        """
+        for ant1, ant2, ind in self._iterate_baselines():
+            lsts = self.data.lst_array[ind]
+            bl_len_m = self.data.uvw_array[ind][
```

Comment:
Lets use the `self.antpos` dictionary instead of this, b/c `uvw_array` can change depending on the phasing of the data...

Reply:
Or maybe `uvw_array` is the correct attribute to use, to account for what phased visibilities actually see on the sky.. not exactly sure about this, and I think it may be dependent on the assumptions of the `model` function. Better to use the `antpos` in the mean time and we can make this question an issue and come back to it later.

Reply:
I don't really understand the subtleties of this. It sounds like it's kind of ambiguous at the moment. If so, maybe it's better to leave it as-is, and once merged, create an issue for it where it can be hashed out in full detail.

Reply:
I think its more safe to use `antpos` over `uvw_array`, b/c `uvw_array` can definitely be changed depending on the phasing of the data, so I think we should default to that for now, and I've created an issue for it in the meantime.. LMK if you have any thoughts why we should default to `uvw_array` over `antpos` for now

Reply:
@jaguirre  agrees. I've changed this in b5a1c00865d817f67b946b2e2a45d233c3961f88.